Remove TransferWhitelist contract

karlb opened the pull request:
The contract is not used anymore. See also https://github.com/celo-org/celo-blockchain/pull/2110.

karlb committed:
Remove TransferWhitelist contract

The contract is not used anymore. See also
https://github.com/celo-org/celo-blockchain/pull/2110.

karlb committed:
cli, sdk: Remove TransferWhitelist

martinvol commented:
I think it's worth noticing that even if we delete the contract itself, our on-chain contract Registry does not have a function to remove, so the deployed implementation will keep being a core-contract. 

I do not see a downside with deleting the contracts, as it is already verified in Celo Explorer: https://explorer.celo.org/mainnet/address/0xb49E4d6F0B7f8d0440F75697E6c8b37E09178BCF/contracts#address-tabs

eelanagaraj commented on packages/protocol/migrationsConfig.js:
```diff
@@ -206,10 +206,6 @@ const DefaultConfig = {
     oracles: [network.from],
     frozen: false,
   },
```
any reason to keep the `transferWhitelist` address list configured below (~L573)?

karlb replied:
I don't remember a reason, so I probably just missed it. It's removed now.

eelanagaraj reviewed (commented):
looks good other than the comment + the GrandaMento CK test failure, which like it's probably caused by the stableToken contract address now being migrated one contract earlier than before ( ithout the transferWhiteList contract migration happening)

karlb committed:
Remove TransferWhitelist from migrationsConfig.js

karlb committed:
Relax contract address check in GrandaMento test

Otherwise the test fails when one of the migrations before the cUSD
deployment changes.

eelanagaraj reviewed (approved):
🚀
edit: flagging that the build is failing, would double check that but once this is updated + build is passing, good to go 👍

karlb commented:
> flagging that the build is failing, would double check that

The current build failures are present in master. See
* https://github.com/celo-org/celo-monorepo/pull/10240#issuecomment-1557060584 (cloud build)
* https://celo-org.slack.com/archives/CP5V8KF51/p1685451495104709 (foundry tests)

I can wait until master is green again and rebase then to get the the tests green.

pahor167 commented on packages/protocol/lib/compatibility/verify-bytecode.ts:
```diff
@@ -15,9 +15,6 @@ import Web3 from 'web3'
 
 let ignoredContracts = [
-  // This contract is not proxied
-  'TransferWhitelist',
```
We should consider leaving this here since it is possible that someone will try to verify previous releases and since this contract is part of previous releases it might happen that verification will fail (or we should double check that it will not fail)

karlb replied:
Keeping the ignore seems the safe thing to do. I've added it back.

karlb committed:
Keep ignore in verify-bytecode.ts

since it is possible that someone will try to verify previous releases
and since this contract is part of previous releases it might happen
that verification will fail (or we should double check that it will not
fail).

See
https://github.com/celo-org/celo-monorepo/pull/10318#discussion_r1211574692.

soloseng reviewed (approved):
🚀
